plugins/org.eclipse.emf.ecore/src/org/eclipse/emf/ecore/plugin/EcorePlugin.java

```diff
@@ -586,13 +603,18 @@ public static Map<String, URI> getEPackageNsURIToDynamicModelLocationMap(boolean
    */
   static public class Implementation extends EclipsePlugin
   {
-    /**
+    
+    
+    private PlainOSGiBundleActivator plainOSGiBundleActivator;
```

Comment:
If I understand the current state correctly IS_ECLIPSE_RUNNING will be false we don't really need this case anymore.  Or?

Reply:
We would still need an additional Activator, in case `IS_ECLIPSE_RUNNING` is false and `IS_OSGI_RUNNING` is true. In this case, we need an activator but we don't have the `EclipsePlugin`. We could copy the stuff from the `PlainOSGiBundleActivator` in here as well, but we would end up with duplicated code.